client/components/deposits-overview/recent-deposits.tsx

```diff
@@ -0,0 +1,122 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+import { Flex, FlexItem, Icon } from '@wordpress/components';
+import { calendar } from '@wordpress/icons';
+
+/**
+ * Internal dependencies.
+ */
+import strings from './strings';
+import './style.scss';
+import DepositStatusChip from 'components/deposit-status-chip';
+import { getDepositDate } from 'deposits/utils';
+import { CachedDeposit } from 'wcpay/types/deposits';
+import { useDeposits } from 'wcpay/data';
+import { formatCurrency } from 'wcpay/utils/currency';
+import { Link } from '@woocommerce/components';
+import { getDetailsURL } from 'wcpay/components/details-link';
+
+interface DepositRowProps {
+	deposit: CachedDeposit;
+}
+interface RecentDeposits {
+	deposits: CachedDeposit[];
+	isLoading: boolean;
+}
+interface RecentDepositsProps {
+	currency: string | undefined;
+}
+
+const tableClass = 'wcpay-deposits-overview__table';
+
+const GetRecentDeposits = ( currency?: string ): RecentDeposits => {
+	const query = {
+		status_is_not: 'estimated',
+		store_currency_is: currency,
+		orderby: 'date',
+		order: 'desc',
+		per_page: '3',
+	};
+	const deposits = useDeposits( query );
+
+	return {
+		deposits: deposits.deposits,
+		isLoading: deposits.isLoading,
+	};
+};
+
+const DepositTableRow: React.FC< DepositRowProps > = ( {
+	deposit,
+}: DepositRowProps ): JSX.Element => {
+	return (
+		<Flex className={ `${ tableClass }__row` }>
+			<FlexItem className={ `${ tableClass }__cell` }>
+				<Icon icon={ calendar } size={ 17 } />
+				<Link href={ getDetailsURL( deposit.id, 'deposits' ) }>
+					{ getDepositDate( deposit ) }
+				</Link>
+			</FlexItem>
+			<FlexItem className={ `${ tableClass }__cell` }>
+				<DepositStatusChip status={ deposit.status } isCompact />
+			</FlexItem>
+			<FlexItem className={ `${ tableClass }__cell` }>
+				{ formatCurrency( deposit.amount, deposit.currency ) }
+			</FlexItem>
+		</Flex>
+	);
+};
+
+/**
+ * Renders the Recent Deposit details component.
+ *
+ * This component includes the recent deposit heading, table and notice.
+ *
+ * @param {DepositRowProps} props Next Deposit details props.
+ * @return {JSX.Element} Rendered element with Next Deposit details.
+ */
+const RecentDeposits: React.FC< RecentDepositsProps > = ( {
```

Comment:
This component's variable name `RecentDeposits` is the same as the `RecentDeposits` TS interface used for the custom hook above.

I'm not exactly sure what impact this has, but I would consider changing either one to avoid confusion.